### DISCUSSION TITLE
[META] Close duplicate issues/proposals

### DISCUSSION THREAD

**Discussion by Unknown6656** [dotnet/csharplang#1317] [General]
I really love this repo for the proposals and discussions, but I have the feeling, that to many issues are duplicates.

In order to "clean up" this repository I would propose to introduce an issue label named something like `[closed - duplicate]` in order tag duplicate issues and close them.

We are currently relying on wonderful users like @ufcpp to find duplicates and while I honour this work, I would also like to remind the community of the following points:
 - This is (normally) the task of an repo moderator
 - Please search for your problem/issue _before_ submitting a new proposal

I can completely understand that the LDT has far more important tasks to do than skimming through GitHub-issues for the search of duplicates, so I propose the following:
Elevate 2..3 active users to the level of moderators, so that they can maintain and cleanup the issues/proposals. I myself would volunteer to do it, as I am browsing this repository on a daily basis.

**Comment by Unknown6656** [comment]
/cc @gafter @MadsTorgersen 
[who do I tag for this?]

**Comment by DavidArno** [comment]
Related: #459 (sorry, couldn't resist 😉)

**Comment by Unknown6656** [comment]
@DavidArno - funny, I remembered the issue you mentioned, but I forgot to link it when I wrote this post.
Thank you^^

**Comment by gafter** [comment]
One problem is that frequently those “duplicate” issues are proposals for a different way to solve the same problem.

**Comment by tannergooding** [comment]
> One problem is that frequently those “duplicate” issues are proposals for a different way to solve the same problem.

Maybe we need another issue to track all the related duplicate issues :trollface:

**Comment by yaakov-h** [comment]
Can we somehow categorize them by problem? Or would that be too difficult?

**Comment by gafter** [comment]
@yaakov-h you mean create a new issue and close the others?

**Comment by yaakov-h** [comment]
@gafter Actually I was thinking of using tags, but the more I think about that the less manageable it seems.

That said, there are a lot of open duplicates that do not provide an alternative solution to the same problem, and it would be worthwhile cleaning those up.

Having more open issues every day - as of now, 861 - makes it harder and harder to find what I'm looking for. The signal to noise ratio is decreasing.

**Comment by orthoxerox** [comment]
> One problem is that frequently those “duplicate” issues are proposals for a different way to solve the same problem.

We can invite people to post their proposal in the "main" issue when closing.

**Comment by DavidArno** [comment]
@orthoxerox,

Currently, that's guaranteed to result in some here complaining to those people that they are muddying the issue and that they should put their ideas in a new issue... ☹️

**Comment by theunrepentantgeek** [comment]
True duplicates should be merged and consolidated - but *different ideas* should be kept separate even if they're attempting to solve similar problems. It's already hard enough to follow the discussion in some of the longer threads, merging discussions of different concepts into one thread would really muddy the waters further.

**Comment by HaloFour** [comment]
I think that the current setup is pretty decent, having members of the community identify and tag the similar/duplicate issues.  The author can then decide whether or not the existing proposals are close enough to warrant moving the discussion and closing the issue.  This seems to scale out fairly well and quite a few of the active members here have participated.  There is certainly some problems with "drive-by" proposals where the author doesn't weigh in.

Worse than the duplicates I think are the problems of so many proposals being seemingly stuck in limbo.  There's a wide gulf between "championed" and "oh Heck no".

**Comment by jnm2** [comment]
> Worse than the duplicates I think are the problems of so many proposals being seemingly stuck in limbo. 

This.

**Comment by Unknown6656** [comment]
> Worse than the duplicates I think are the problems of so many proposals being seemingly stuck in limbo. There's a wide gulf between "championed" and "oh Heck no".

I definitely agree with this. We would either need some tags like mentioned in #459 or some kind of "voting system" used by the senior devs or LDT representatives/core members, so that they can easily indicate whether a submitted issue is worth the effort/reward or not.

**Comment by iam3yal** [comment]
In my opinion voting is a bad idea for the task, I really don't think we need to make it any more complex than it is, however, it can certainly be improved, the LDT can mark posts with "canonical" and have a bot closing issues when specific community members links to it using a specific syntax, e.g, "!!dup #link", this frees the LDT from doing it and it also gives specific control to some people without having a moderator status of some sort.

Just to clarify only proposals should be tagged with "canonical" and if there are two proposals that are tagged with "canonical" and one of them is a dup of the other the bot can send a message to the LDT to resolve this conflict so in this case it wouldn't close the dups until there's one canonical issue.

This is just an idea, it might work but it may also not worth the efforts, I don't know, :)

**Comment by yaakov-h** [comment]
@eyalsk That seems nicely simple, and straightforward to implement, too.

**Comment by quinmars** [comment]
> One problem is that frequently those “duplicate” issues are proposals for a different way to solve the same problem.

I agree that it is important to keep competing proposal in seperated threads. That makes it easier to reference those proposals. For example, proposal #133 and #140 are similiar in their intend. They are, however, different enough that I can say that I prefer #140. But some proposal are real duplicates, for example #1420 is an exact duplicate of #133.

Of course there is a gray area between real competing proposals and exact duplicates. Maybe someone (a volunteer as proposed by @Unknown6656 ) can tag those issues, that tend to belong to the latter, as "possible duplicate". This gives the proposal poster the opportunity to work out the differences and making it a stronger competing proposal. If the arguments are not compelling the issue could be closed at a later point.
